### PR TITLE
[JENKINS-68904] Adress ESLint violations

### PR DESCRIPTION
**Pull request by NotMyFault**
Addresses the ESLint part of [JENKINS-68904](https://issues.jenkins.io/browse/JENKINS-68904)
The latest commit simulates a CI failure and will be reverted once demonstrated.

### Proposed changelog entries

* Entry 1: Issue, Human-readable Text
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6794"></a>

### CODE REVIEW

**Review by timja — APPROVED** [review]

File: war/src/main/js/filter-build-history.js
Comment (basil) [inline war/src/main/js/filter-build-history.js:334]: Ah yes, the intricacies of variable scope in JavaScript… :smile: I don't understand this but I trust you.

File: war/src/main/js/filter-build-history.js
Comment (basil) [inline war/src/main/js/filter-build-history.js:363]: Ah yes, the intricacies of variable scope in JavaScript… :smile: I don't understand this but I trust you.

**Review by basil — APPROVED** [review]
commit 6b70fc81111f52473b91d5a4f933affaf443fcc0 is not fully reverted by commit f40fca8cd3922495027bd1ecd6ece8e6a3c2e755. I would prefer that it be fully reverted and that we discuss the topic of enforcement in a separate PR (the fix for JENKINS-68903), since there are multiple ways to implement JENKINS-68903 and the partial revert in f40fca8cd3922495027bd1ecd6ece8e6a3c2e755 implements the solution to JENKINS-68903 that I describe in that bug as "less attractive."

**Comment by NotMyFault** [comment]
The commit has been reverted properly 👍🏻

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
